src/Serialization/main.jl

```diff
@@ -204,6 +205,57 @@ function load_internal_generic(s::DeserializerState, ::Type{T}, dict::Dict) wher
     return T(fields...)
 end
 
+################################################################################
+# Include internal saves for specific types
+
+include("basic_types.jl")
+include("containers.jl")
+include("PolyhedralGeometry.jl")
+include("Combinatorics.jl")
+include("Fields.jl")
+include("ToricGeometry.jl")
+include("Rings.jl")
+include("polymake.jl")
+include("TropicalGeometry.jl")
+include("QuadForm.jl")
+
+################################################################################
+# Include upgrade scripts
+
+include("upgrades/main.jl")
+
+################################################################################
+# Loading with upgrade checks on dict
+
+# Finds the first version where an upgrade can be applied and then incrementally
+# upgrades to the current version
+function upgrade(dict::Dict{Symbol, Any}, dict_version::VersionNumber)
+    upgraded_dict = dict
+    for upgrade_script in upgrade_scripts
+        script_version = version(upgrade_script)
+
+        if dict_version < script_version
+            println("upgrading serialized data to version $script_version")
```

Comment:
I think this message can become a problem when loading huge data sets. Perhaps use `@info` or `@logmsg` or one of the logging macros from Hecke (about which I know little but @fieker or @thofma can certainly suggest something).

The key point is that there should be a way for the user resp. calling code to suppress this message.

(This can of course be added in a future PR, in which case I'd suggest to just add a TODO/FIXME comment here)